builtin/credential/aws-ec2/path_login.go

```diff
@@ -265,60 +320,97 @@ func (b *backend) parseIdentityDocument(s logical.Storage, pkcs7B64 string) (*id
 // option is enabled on the registered role.
 func (b *backend) pathLoginUpdate(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	identityDocB64 := data.Get("identity").(string)
```

Comment:
Yeah. It could just be a `[]byte`. Changed it. Should we make base64 encoding an optional thing to do, for identity document?

You are right about signature is received base64 from the metadata endpoint.